revolv/project/views.py

```diff
@@ -155,20 +157,64 @@ def get_context_data(self, **kwargs):
         context['GOOGLEMAPS_API_KEY'] = settings.GOOGLEMAPS_API_KEY
         return context
 
+class PostProjectUpdateView(UserDataMixin, UpdateView):
+    model = Project
+    template_name = 'project/post_project_update.html'
+    form_class = forms.PostProjectUpdateForm
 
-class PostFundingUpdateView(UpdateView):
-    """
-    The view to send out post funding updates about a project after it has completed.
+    def dispatch(self, request, *args, **kwargs):
+        self.user = request.user
+        self.is_authenticated = self.user.is_authenticated()
+        if self.is_authenticated:
+            self.user_profile = RevolvUserProfile.objects.get(user=self.user)
+            self.is_ambassador = self.user_profile.is_ambassador()
 
-    Accessed through /project/{project_id}/update
-    """
-    model = Project
-    template_name = 'project/post_funding_update.html'
-    form_class = forms.PostFundingUpdateForm
+        if not self.is_ambassador:
+            return self.deny_access()
+        return super(PostProjectUpdateView, self).dispatch(request, args, kwargs)
 
     def get_success_url(self):
-        return reverse('project:view', kwargs={'pk': self.get_object().id})
+        return reverse('project:review', kwargs={'pk': self.get_object().id})
+
+    def form_valid(self, form):
+        text = form.cleaned_data['update_text']
+        project = self.get_object()
+        project.add_update(text)
+        return super(PostProjectUpdateView, self).form_valid(form)
+
+    def form_invalid(self, form):
+        print("oh no", form)
+
+class EditProjectUpdateView(UserDataMixin, UpdateView):
+    
+    model = ProjectUpdate
+
+    #STILL HAVE TO MAKE THIS TEMPLATE AND FORM
```

Comment:
I think we don't really need a comment here because it's pretty self explanatory why the attributes of this view are what they are. However, you can leave the comment in if you expand on why it's not self explanatory why you might need the template and form there.